### PR TITLE
Bug Fix in New-AzManagementGroup cmdlet

### PR DESCRIPTION
**Pull request by git-thomasdolan**
## Description

Fixed a bug in the New-AzManagementGroup cmdlet raised by [this](https://github.com/Azure/azure-powershell/issues/16704#issuecomment-1278645100) issue. Our API returns a 202 when creating a new management group and an async operation as the result instead of the MG. The cmdlet was trying to cast this async header as a ManagementGroup and it would throw this error: `Unable to cast object of type 'Newtonsoft.Json.Linq.JObject' to type 'Microsoft.Azure.Management.ManagementGroups.Models.ManagementGroup'`. Doing a GET if a 202 is returned fixed this issue

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by BethanyZhou** [comment]
Hi @git-thomasdolan , your changes make many test cases failed. one of them is 
```
 [xUnit.net 00:00:24.02]     Microsoft.Azure.Commands.Resources.Test.ScenarioTests.ManagementGroupsTests.TestNewManagementGroup [FAIL]
EXEC : error Message:  [/Users/runner/work/1/s/build.proj]
     System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Unable to cast object of type 'Microsoft.Azure.Management.ManagementGroups.Models.ManagementGroup' to type 'Newtonsoft.Json.Linq.JObject'.
   ``` 
please have a look

**Comment by git-thomasdolan** [comment]
> Hi @git-thomasdolan , your changes make many test cases failed. one of them is
> 
> ```
>  [xUnit.net 00:00:24.02]     Microsoft.Azure.Commands.Resources.Test.ScenarioTests.ManagementGroupsTests.TestNewManagementGroup [FAIL]
> EXEC : error Message:  [/Users/runner/work/1/s/build.proj]
>      System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Unable to cast object of type 'Microsoft.Azure.Management.ManagementGroups.Models.ManagementGroup' to type 'Newtonsoft.Json.Linq.JObject'.
> ```
> 
> please have a look

I did not commit the changes to the Scenario Test Recordings so that was likely why the Tests failed. I just committed those changes and hopefully all the Tests in the Pipeline pass now

**Comment by BethanyZhou** [comment]
**SHOULD update ChangeLog.md file(s) appropriately**
For any service, the ChangeLog.md file can be found at src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md
A snippet outlining the change(s) made in the PR should be written under the ## Upcoming Release header in the past tense. Add changelog in description section if PR goes into [generation](https://github.com/Azure/azure-powershell/tree/generation) branch.

BTW, I saw a commit in past year, could you clean your commit history and make sure there are only needed for this bug fix? One way is resetting your branch with upstream main and cherry-pick your commits
image

**Comment by BethanyZhou** [comment]
@git-thomasdolan Could you clean your commit history please? Your PR carries commits from last year and others.

**Comment by git-thomasdolan** [comment]
Sorry, took me a bit to get the history cleaned up correctly. The checks passed now so let me know if there is anything else I need to do

File: src/Resources/Resources/ChangeLog.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 
 ## Upcoming Release
 * Updated behavior of Get-AzPolicyDefinition which previously returned all definitions when -Id was provided with a nonexistent policy definition id. Fixed to correctly throw a 404 exception in this case.
+* Fixed an issue when running the `New-AzManagementGroup` command where it tried to cast an async operation as a Management Group. [#21000]
```
Comment (BethanyZhou) [inline src/Resources/Resources/ChangeLog.md:23]:
```suggestion
* Fixed an issue when running the `New-AzManagementGroup` command where it tried to cast an async operation as a Management Group. [#21000]
* Updated behavior of Get-AzPolicyDefinition which previously returned all definitions when -Id was provided with a nonexistent policy definition id. Fixed to correctly throw a 404 exception in this case.
```

**Review by BethanyZhou — DISMISSED** [review]

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**